sql/log_event_server.cc

```diff
@@ -5803,6 +5803,25 @@ int Rows_log_event::do_add_row_data(uchar *row_data, size_t length)
 
 #if defined(HAVE_REPLICATION)
 
+void Rows_log_event::make_thd_info(thd_info_cache cache,
+  const char *class_name, std::initializer_list<const char *> method_names)
```

Comment:
For the `main` branch, an alternative to a list of step names is [a list of steps themselves](https://refactoring.guru/design-patterns/chain-of-responsibility).